src/infra/core/tests/tests/repos/test_dataset_repository_shared.rs

```diff
@@ -61,6 +61,62 @@ pub async fn test_create_dataset(repo: &dyn DatasetRepository, account_name: Opt
 
 /////////////////////////////////////////////////////////////////////////////////////////
 
+pub async fn test_create_and_get_case_insensetive_dataset(
+    repo: &dyn DatasetRepository,
+    account_name: Option<AccountName>,
+) {
+    let dataset_alias_to_create =
```

Comment:
Please extend this test to create two datasets `account/foo` and `ACCOUNT/BAR` and then try accessing them as `AcCoUnT/Foo` and `AcCoUnT/Boo`.

I have a suspicion that during the creation process we don't normalize the account name correctly and the datasets will be created under two separate account directories (i.e. we will end up with both `account` and `ACCOUNT` dirs)

Reply:
Extended